kong/runloop/certificate.lua

```diff
@@ -371,11 +374,162 @@ local function get_ca_certificate_store(ca_ids)
 end
 
 
+-- ordinary entities that reference ca certificates
+-- the value denotes which cache (kong.cache or kong.core_cache) is used
+local CA_CERT_REFERENCE_ENTITIES = {
+  "services",
+}
+
+-- plugins that reference ca certificates
+-- Format:
+-- mtls-auth = true
+local CA_CERT_REFERENCE_PLUGINS = {
+}
```

Comment:
I think it's enough to define the types of `CA_CERT_REFERENCE_PLUGINS` and `reference_plugins` as arrays, which has the advantage that, assuming we're deleting a ca_certificate referenced by more than one plugin at the same time, the error descriptions returned by the admin api will always be fixed instead of a variable one (due to the unordered nature of luajit `pairs`).

Reply:
Agree. Previously we needed to retrieve the table by the plugin name in `gen_iterator` thus defining them as maps. Now that we have changed the implementation of `gen_iterator` to use the filter, defining them as arrays is sufficient.